lib/HLSL/DxilValidation.cpp

```diff
@@ -2346,6 +2369,67 @@ static void ValidateDxilOperationCallInProfile(CallInst *CI,
           {"CreateHandle", "Shader model 6.5 and below"});
     }
     break;
+
+  case DXIL::OpCode::ThreadId: { // SV_DispatchThreadID
+    if (shaderKind != DXIL::ShaderKind::Node) {
+      break;
+    }
+
+    if (nodeLaunchType == DXIL::NodeLaunchType::Broadcasting) {
+      // good
+    } else {
+      ValCtx.EmitInstrFormatError(CI,
+                                  ValidationRule::InstrSVConflictingLaunchMode,
+                                  {"ThreadId", "SV_DispatchThreadID",
+                                   GetLaunchTypeStr(nodeLaunchType)});
+    }
+  } break;
```

Comment:
Another formatting nitpick - `break` on the same line as a closing brace looks weird.

Reply:
Clang format will place, and continue replacing, the break  on the same line as the closing brace. I think I'll leave this as is.

Reply:
Oh, this break should probably be inside the block for the label, no?
```c++
case abc: {
  // ...
  break;
}
case xyz: ...
```
rather than
```c++
case abc: {
  // ...
}
break;
case xyz: ...
```

Reply:
Actually, why are we using braces for these case labels at all? They don't need to introduce a scope.